manifests/dcgm/telegraf.pp

```diff
@@ -0,0 +1,131 @@
+  Integer $dcgm_telegraf_port,
+  Integer $dcgm_telegraf_py_port,
+  Boolean $enable,
+) {
+
+  if ($enable) {
+    $ensure_parm = 'present'
+  } else {
+    $ensure_parm = 'absent'
+  }
+
+  #
+  # Setup nvidia-dcgm systemd service
+  #
+  systemd::unit_file { 'nvidia-dcgm.service':
+    content => file("${module_name}/nvidia-dcgm.service"),
+    enable  => $enable,
+    active  => $enable,
+    before  => Systemd::Unit_File['dcgmd-telegraf.service'],
+    require => Package['datacenter-gpu-manager'],
+  }
+
+  #
+  # Setup dcgmd-telegraf
+  #
+
+  # Setup variables
+  if find_file('/usr/bin/python3') {
+    $exec_start = '/usr/bin/python3'
+    $dcgm_telegraf_py_path = '/usr/local/dcgm/bindings/python3/dcgm_telegraf.py'
+  } elsif find_file('/usr/bin/python') {
+    $exec_start = '/usr/bin/python'
+    $dcgm_telegraf_py_path = '/usr/local/dcgm/bindings/dcgm_telegraf.py'
+  } else {
+    fail('Unable to determine python version')
+  }
+
+  # TMP fix in place for some bug in NVIDIA DCGM
+  file_line { 'fix_dcgm_telegraf_py':
+    path               => $dcgm_telegraf_py_path,
+    match              => '        self\.m_sock.sendto\(payload, self\.m_dest\)',
+    line               => '        self.m_sock.sendto(payload.encode(), self.m_dest)',
+    append_on_no_match => 'false',
+    require            => Package['datacenter-gpu-manager'],
+    notify             => Service['dcgmd-telegraf.service'],
+  }
+
+  # First Modification so dcgmd-telegraf listens on a static port and only on localhost
+  file_line { 'dcgm_telegraf_py_localhost_listen_only':
+    path               => $dcgm_telegraf_py_path,
+    after              => 'DEFAULT_TELEGRAF_PORT = 8094',
+    # Cannot add them both in one go, line isn't smart enough to check multiple lines
+    # So you instead end up added the two lines every time puppet runs
+    line               => "LISTEN_HOST = '127.0.0.1'",
+    append_on_no_match => 'false',
+    require            => Package['datacenter-gpu-manager'],
+    notify             => Service['dcgmd-telegraf.service'],
+  }
+
+  # Second Modification so dcgmd-telegraf listens on a static port and only on localhost
+  file_line { 'dcgm_telegraf_py_localhost_listen_only_part2':
+    path               => $dcgm_telegraf_py_path,
+    after              => "LISTEN_HOST = '127.0.0.1'",
+    line               => "LISTEN_PORT = ${dcgm_telegraf_py_port}",
+    append_on_no_match => 'false',
+    require            => [ Package['datacenter-gpu-manager'], File_Line['dcgm_telegraf_py_localhost_listen_only'] ],
+    notify             => Service['dcgmd-telegraf.service'],
+  }
+
+  # Third Modification so dcgmd-telegraf listens on a static port and only on localhost
+  file_line { 'dcgm_telegraf_py_localhost_listen_only_part3':
+    path               => $dcgm_telegraf_py_path,
+    after              => '        self\.m_sock = socket\(AF_INET, SOCK_DGRAM\)',
+    line               => '        self.m_sock.bind((LISTEN_HOST, LISTEN_PORT))',
```

Comment:
same missing a `match`.

Reply:
That `file_line` is adding a new line instead of replacing an existing line. So the `after` param takes the place of a `match` param